mutters-core/src/main/java/com/rabidgremlin/mutters/core/CompoundSlot.java

```diff
@@ -35,26 +36,28 @@ public CompoundSlot(String name, Slot firstSlot, Slot secondSlot)
 
   /*
    * (non-Javadoc)
-   * 
-   * @see com.rabidgremlin.mutters.core#match(String token, Context context)
-   * 
+   *
+   * @see com.rabidgremlin.mutters.core.Slot#match(String token, Context context)
+   *
    */
+  @SuppressWarnings("unchecked")
   @Override
-  public SlotMatch match(String token, Context context)
+  public SlotMatch<T> match(String token, Context context)
   {
-    SlotMatch match = firstSlot.match(token, context);
+    SlotMatch<? extends T> match = firstSlot.match(token, context);
     if (match == null)
     {
       match = secondSlot.match(token, context);
     }
 
-    return match;
+    // this cast is type safe because SlotMatch is immutable, in particular the 'value' field
+    return (SlotMatch<T>) match;
```

Comment:
Even if it was mutable, it would still be safe as it is an upcast right?

Reply:
Yeah, it wouldn't fail (at compile or run time). But it might not make sense.

So if ```SlotMatch``` had a method ```setValue``` and you did this:
```
    CompoundSlot<Object> compound = new CompoundSlot<>("compound", new LiteralSlot("literal"), new NumberSlot("number"));
    SlotMatch<Object> slotMatch = compound.match("hey", null);
    slotMatch.setValue(2.5d);
```
Now have a slot match with ```LiteralSlot``` (which matches ```String```) and ```Double``` value.
But doesn't fail because the ```SlotMatch``` is defined by ```Object```.


..Thinking the return type should be ```SlotMatch<? extends T>```... (even though they say not to use wildcard in return type)

Reply:
Hmm.. if return type is ```SlotMatch<? extends T>``` always have to declare ```SlotMatch``` with wildcards (or use raw type), maybe best to leave it out since it will never have a ```setValue``` method.